### ISSUE TITLE
missing typecheck error when adding to sets with wrong value type

### ISSUE THREAD

**Issue by psybers** [boalang/compiler#253]
````
o: output sum of int;
o << 1;

s: set of string;
cf: ChangedFile;
add(s, cf);
````

This code should give an error in type checker.  Instead it gets to codegen, where we get this error:

````
BoaWeb137270686620821364.java:143: cannot find symbol
symbol  : method add(boa.types.Diff.ChangedFile)
location: class java.util.HashSet
				___s.add(___cf);
				    ^
````

**Comment by psybers** [comment]
This seems to be fixed in master already.

```
t.boa: compilation failed: Encountered typecheck error at line 6, column 0. no such function add([set of string, ChangedFile])
add(s, cf);
^^^
	at boa.compiler.visitors.TypeCheckingVisitor.visit(TypeCheckingVisitor.java:422)
```
